pilot/pkg/xds/cds.go

```diff
@@ -39,7 +38,12 @@ var skippedCdsConfigs = map[config.GroupVersionKind]struct{}{
 	gvk.Secret:                {},
 }
 
-func cdsNeedsPush(req *model.PushRequest) bool {
+// Map all configs that impacts CDS for gateways.
+var pushCdsGatewayConfig = map[config.GroupVersionKind]struct{}{
+	gvk.VirtualService: {},
```

Comment:
We also need Gateway. If I add a new gateway it may add to my workload and VS may bind to it, giving new clusters

Reply:
Good point